Content.Server/Construction/AnchorableSystem.cs

```diff
@@ -104,6 +113,27 @@ private void OnAnchorComplete(EntityUid uid, AnchorableComponent component, TryA
             );
         }
 
+        private bool CheckUnstackable(EntityUid uid, EntityUid user, EntityCoordinates coordinates)
+        {
+            if (_tagSystem.HasTag(uid, "Unstackable"))
+            {
+                // Need to enforce the unstackable rules on anchoring also.
+                if (NoUnstackableInTile.AnyUnstackableTiles(coordinates, _tagSystem))
+                {
+                    var message = NoUnstackableInTile.GuidebookString;
+                    if (message != null)
+                    {
+                        // Show the reason to the user:
+                        _popup.PopupEntity(Loc.GetString(message), uid, user);
+                    }
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
This condition looks the opposite of what I'd expect, the other one is a check and returns false if you can't do it but this returns true if you can't.

Also put it on the shared system and use popupclient (and same on the bool).

Reply:
Ah, that's what you meant before about calling into the anchorable system. Will do. 

You're right, not only is the condition flipped, the name of the function is vague.  Something like `DoesUnstackableCancel` or similar would be a much better name. If I renamed it to that, the condition would then be OK.

Reply:
I don't remember the technical name for it but as long as it's a "positive" condition i.e. "DoesX" rather than "DoesNotX" kinda style.